### DISCUSSION TITLE
Reorg job updating stats

### DISCUSSION THREAD

**Discussion by teknogrl** [olahallengren/sql-server-maintenance-solution#592] [Questions & Answers]
Hello, 
We are using your reorg/reindex jobs in production.  I have the following set up, which I believe means it would not update stats, however in the commandlog table I'm seeing a rebuild of the index immediately following the reorg of the index, screen shot attached.  Am I misunderstanding the use of the word NULL on update stats?
![reorg_with_updatestats](https://user-images.githubusercontent.com/75396098/141526285-5c7d1872-de14-4c8b-84b7-b55ed772f0a9.png)

EXECUTE [dbo].[IndexOptimize]
@Databases = 'DATABASENAME',
@Indexes = 'DATABASENAME.dbo.%',
@LogToTable = 'Y',
@LOBCompaction = 'N',
@FragmentationMedium = 'INDEX_REORGANIZE',
@FragmentationHigh = NULL,
@UpdateStatistics = NULL

**Comment by BHubartt38** [comment]
I would suggest commenting out or removing the parameter @UpdateStatistics.

I can't remember exactly where I read it, but I created a separate job to update statistics for better control.

You can try the following.  If you want to test you can change @LogToTable and @Execute to 'N' to see what will be executed.
-- Statistics Only
EXECUTE dbo.IndexOptimize
	@Databases = @DatabasesToBackup,
	@FragmentationLow = NULL,
	@FragmentationMedium = NULL,
	@FragmentationHigh = NULL,
	@UpdateStatistics = 'ALL',
	@OnlyModifiedStatistics = 'Y',
	@LogToTable = 'Y',
	@Execute = 'Y'

-- Indexes Only
EXECUTE dbo.IndexOptimize
@Databases = @DatabasesToBackup,
@FragmentationLow = NULL,
@FragmentationMedium = 'INDEX_REORGANIZE,INDEX_REBUILD_ONLINE',
@FragmentationHigh = 'INDEX_REBUILD_ONLINE',
@LogToTable = 'Y',
@Execute = 'Y';

**Reply by teknogrl** [reply]
Thanks for the response, the job actually had the updates stats excluded, and we were still seeing update stats happening, (and being logged to hte table) so I updated the job to include the update stats but with the NULL value.  Any other thoughts?

**Reply by BHubartt38** [reply]
Hmmm...strange...

...but, I think the reason is tied to @Fragmentation<Low/Medium/High>

If you do a text search of the procedure for "Update statistics?", so somewhere around line 2015, there is some logic that sets the parameter @CurrentUpdateStatistics.  I'm running Enterprise edition so I'm able to use INDEX_REBUILD_ONLINE which causes @CurrentUpdateStatistics to be set to 'N' which I think does not do a statistics update after the index update.
